internal/jimm/cloud.go

```diff
@@ -213,15 +213,43 @@ func (j *JIMM) AddHostedCloud(ctx context.Context, user *openfga.User, tag names
 	if cloud.HostCloudRegion == "" {
 		return errors.E(op, errors.CodeCloudRegionRequired, "cloud host region not specified")
 	}
-	parts := strings.SplitN(cloud.HostCloudRegion, "/", 2)
-	if len(parts) != 2 || parts[0] == "" {
-		return errors.E(op, errors.CodeBadRequest, fmt.Sprintf("invalid cloud/region format %q", cloud.HostCloudRegion))
-	}
-	region, err := j.Database.FindRegion(ctx, parts[0], parts[1])
-	if errors.ErrorCode(err) == errors.CodeNotFound {
-		return errors.E(op, err, errors.CodeNotFound, fmt.Sprintf("unable to find cloud/region %q", cloud.HostCloudRegion))
-	} else if err != nil {
-		return errors.E(op, err)
+
+	var region *dbmodel.CloudRegion
+	var err error
+	if !strings.Contains(cloud.HostCloudRegion, "/") {
+		cl := dbmodel.Cloud{
+			Name: cloud.HostCloudRegion,
+		}
+		if err := j.Database.GetCloud(ctx, &cl); err != nil {
+			return errors.E(op, errors.CodeBadRequest, "unable to find cloud %q", cloud.HostCloudRegion)
+		}
+		if len(cl.Regions) > 1 {
+			return errors.E(op, errors.CodeBadRequest, "unable to determine a unique cloud region for cloud %q", cloud.HostCloudRegion)
+		}
+		region = &cl.Regions[0]
+	} else {
+		parts := strings.Split(cloud.HostCloudRegion, "/")
+		if len(parts) != 2 || parts[0] == "" {
+			return errors.E(op, errors.CodeBadRequest, fmt.Sprintf("invalid cloud/region format %q", cloud.HostCloudRegion))
+		}
+
+		findRegionFunctions := []func(context.Context, string, string) (*dbmodel.CloudRegion, error){
+			j.Database.FindRegionByCloudType,
+			j.Database.FindRegionByCloudName,
```

Comment:
Technically this will cause an issue if you have two clouds, where the name of one cloud is the same as the "type" of the other. I don't know where the cloud types come from through?
E.g. 
Cloud 1: 
Type=k8s
Name=myCloud

Cloud 2:
Type=foo
Name=k8s

Maybe this situation is impossible/unlikely.

Reply:
juju has a cloud type defined per provider in their code..